<a href="https://colab.research.google.com/github/dmitry-petrov-dev/prtheory_mathstat/blob/lesson2/HW_Lesson_2_%D0%94%D0%B8%D1%81%D0%BA%D1%80%D0%B5%D1%82%D0%BD%D1%8B%D0%B5_%D1%81%D0%BB%D1%83%D1%87%D0%B0%D0%B9%D0%BD%D1%8B%D0%B5_%D0%B2%D0%B5%D0%BB%D0%B8%D1%87%D0%B8%D0%BD%D1%8B_%D0%97%D0%B0%D0%BA%D0%BE%D0%BD_%D1%80%D0%B0%D1%81%D0%BF%D1%80%D0%B5%D0%B4%D0%B5%D0%BB%D0%B5%D0%BD%D0%B8%D1%8F_%D0%B2%D0%B5%D1%80%D0%BE%D1%8F%D1%82%D0%BD%D0%BE%D1%81%D1%82%D0%B5%D0%B9_%D0%91%D0%B8%D0%BD%D0%BE%D0%BC%D0%B8%D0%B0%D0%BB%D1%8C%D0%BD%D1%8B%D0%B9_%D0%B7%D0%B0%D0%BA%D0%BE%D0%BD_%D1%80%D0%B0%D1%81%D0%BF%D1%80%D0%B5%D0%B4%D0%B5%D0%BB%D0%B5%D0%BD%D0%B8%D1%8F_%D0%A0%D0%B0%D1%81%D0%BF%D1%80%D0%B5%D0%B4%D0%B5%D0%BB%D0%B5%D0%BD%D0%B8%D0%B5_%D0%9F%D1%83%D0%B0%D1%81%D1%81%D0%BE%D0%BD%D0%B0.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**1. Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.**

In [ ]:
from math import factorial
import numpy as np

In [ ]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [ ]:
def binom (n, k ,p):
  return combinations(n, k) * (p ** k) * (1 - p) ** (n - k)

$$p(k=85) = C_n^kp^kq^{n-k} = C_{100}^{85}p^{85}q^{15} = \frac{100!}{85!(100-85)!}

\cdot 0.8^{85} \cdot 0.2^{15}$$

In [ ]:
n = 100
k = 85
p = 0.8
X = binom(n, k, p)
X

0.048061793700746355

Ответ: P = 0.048

**2. Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек.**

**Какова вероятность, что ни одна из них не перегорит в первый день?**

In [ ]:
def puassone(n, m, p):
  lambda_ = n * p
  return (lambda_**m / np.math.factorial(m)) * np.exp(-lambda_)

In [ ]:
n = 5000
m = 0
p = 0.0004

In [ ]:
P = puassone(n, m, p)
P

0.1353352832366127

Ответ: P = 0.135

 Какова вероятность, что перегорят ровно две?

In [ ]:
n = 5000
m = 2
p = 0.0004

In [ ]:
P = puassone(n, m, p)
P

0.2706705664732254

Ответ: P = 0.27

**3. Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?**

In [ ]:
n = 144
k = 70
p = 1/2

In [ ]:
P = binom(n, k, p)
P

0.06281178035144776

Ответ: P = 0.628

**4. В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча.**

**Какова вероятность того, что все мячи белые?**

In [ ]:
b1 = 10
b1_w = 7
b2 = 11
b2_w = 9
k = 2

In [ ]:
P1 = combinations(b1_w, k) / combinations(b1, k)
P1

0.4666666666666667

In [ ]:
P2 = combinations(b2_w, k) / combinations(b2, k)
P2

0.6545454545454545

In [ ]:
P = P1 * P2
P

0.3054545454545455

Ответ: P = 0.305

 **Какова вероятность того, что хотя бы один мяч белый?**

In [ ]:
# вероятность вытащить не белый мяч из ящика 1
P1 = combinations(3, 2) / combinations(10, 2)
P1

0.06666666666666667

In [ ]:
# вероятность вытащить не белый мяч из ящика 2
P2 = combinations(2, 2) / combinations(11, 2)
P2

0.01818181818181818

In [ ]:
P = 1 - P1*P2
P

0.9987878787878788

Ответ: P = 0.99(87)

**Какова вероятность того, что ровно два мяча белые?**

In [ ]:
#     BOX1            | BOX2
# (1) БелыйБелый      | НеБелыйНебелый
# (2) БелыйНеБелый    | БелыйНеБелый
# (3) БелыйНеБелый    | НеБелыйБелый
# (4) НеБелыйБелый    | БелыйНеБелый
# (5) НеБелыйБелый    | НеБелыйБелый
# (6) НеБелыйНеБелый  | БелыйБелый

P1 = (7/10) * (6/9) * (2/11) * (1/10)
P2 = (7/10) * (3/9) * (9/11) * (2/10)
P3 = (7/10) * (3/9) * (2/11) * (9/10)
P4 = (3/10) * (7/9) * (9/11) * (2/10)
P5 = (3/10) * (7/9) * (2/11) * (9/10)
P6 = (3/10) * (2/9) * (9/11) * (8/10)

In [ ]:
P = P1 + P2 + P3 + P4 + P5 + P6
P

0.20484848484848484

Ответ: P = 0.20(48)